client/components/steps/woocommerce.jsx

```diff
@@ -26,49 +31,39 @@ module.exports = React.createClass( {
 
 	_onChange: function() {
 		this.setState( getJetpackState() );
+		if ( this.state.wooCommerceStatus && this.state.redirect_to_woocommerce_setup ) {
+			window.setTimeout( this.goToWooSetup, 10 );
+		}
 	},
 
 	getInitialState: function() {
-		let state = getJetpackState();
-
-		const { install_woo } = JPS.bloginfo;
-		let business_name = JPS.bloginfo.business_name;
-		if ( 'undefined' === typeof business_name ) {
-			business_name = state.site_title;
-		}
+		return getJetpackState();
+	},
 
-		state = Object.assign( {}, state, { business_name, install_woo } );
-		return state;
+	goToWooSetup: function() {
+		jQuery(window).off('beforeunload');
```

Comment:
Here, we are creating an alert that pops up when someone leaves the JPO flow before finishing: https://github.com/Automattic/jetpack-onboarding/blob/master/client/stores/data-store.js#L37

Without turning off the handler, we will see this alert when we send folks to woo setup